networkd/network_monitor_thread.ml

```diff
@@ -299,13 +299,28 @@ let ip_watcher () =
     end;
     loop ()
   in
-  while true do
-    try
-      info "(Re)started IP watcher thread";
-      loop ()
-    with e ->
-      warn "Error in IP watcher: %s\n%s" (Printexc.to_string e) (Printexc.get_backtrace ())
-  done
+  try
+    while true do
+      try
+        info "(Re)started IP watcher thread";
+        loop ()
+      with e -> begin
+        warn "Error in IP watcher: %s\n%s" (Printexc.to_string e) (Printexc.get_backtrace ());
+        match !watcher_pid with
+        | None -> raise e
+        | Some pid ->
+          let quited, _ = Forkhelpers.waitpid_nohang pid in
+          if quited <> 0 then begin
+            warn "address monitoring process quitted, try to restart it";
+            raise e;
```

Comment:
In OCaml, semicolons are used in between sequential statements, and are not needed after the last one in a block. I.e this line, and the two lines below, do not need semicolons.